packages/plugin-session-replay-browser/src/session-replay.ts

```diff
@@ -38,12 +44,36 @@ class SessionReplay implements SessionReplayEnrichmentPlugin {
   maxPersistedEventsSize = MAX_EVENT_LIST_SIZE_IN_BYTES;
   interval = MIN_INTERVAL;
   timeAtLastSend: number | null = null;
+  options: SessionReplayOptions;
+  shouldRecord = true;
+
+  constructor(options?: SessionReplayOptions) {
+    this.options = { ...options };
+  }
 
   async setup(config: BrowserConfig) {
     config.loggerProvider.log('Installing @amplitude/plugin-session-replay.');
 
     this.config = config;
+    this.config.sessionId = config.sessionId;
     this.storageKey = `${STORAGE_PREFIX}_${this.config.apiKey.substring(0, 10)}`;
+
+    if (typeof config.defaultTracking === 'boolean') {
```

Comment:
Why do we have this handling of type which implies it's handling a type for backwards compatibility

Reply:
defaultTracking can be of either type boolean or an object: https://www.docs.developers.amplitude.com/data/sdks/browser-2/#configuration